### ISSUE TITLE
How to use seed to guarantee reproducibility for only single environment?

### ISSUE THREAD

**Issue by Zebin-Li** [aai-institute/tianshou#546]
tianshou version 0.4.5 64 bit (AMD64)] win32 python 3.8

Hello,

I was trying to perform RL algorithm in a customized environment. To ensure reproducibility, I added 
```
def seed(self, seed):
    np.random.seed(seed)
```
in the environment, and called `env.seed(123)` when training. The code is shown below:

```
net = Net(state_shape, action_shape, hidden_sizes = [256, 128, 128])
optim = torch.optim.Adam(net.parameters(), lr=1e-3)
policy = ts.policy.DQNPolicy(net, optim, discount_factor=0.99, estimation_step=1, target_update_freq=10)
replay_buffer = ReplayBuffer(size=500)
policy.set_eps(0.1)
    for i in trange(5):
        env.seed(123)
        obs, terminal = env.reset(), False
            while not terminal:
                batch = Batch(obs=[obs], info=None)
                act = policy(batch).act[0]
                replay_buffer.add(Batch(obs=[obs], act=act, obs_next=[next_obs], rew=reward, done=terminal, info=None))
                obs = next_obs
                policy.update(10, buffer=replay_buffer)
```
But it did not work. Could you please provide some hints about it? thank you very much!

**Comment by Trinkle23897** [comment]
Have you tried to add `np.random.seed(123)` and `torch.manual_seed(123)` at the very beginning? Also you need to add `act = policy.exploration_noise(act)` after `act = policy(batch).act[0]` to activate epsilon-greedy.

**Comment by Zebin-Li** [comment]
Thank you for the quick response! It solved!
